WiseBuy/View Controllers/DealsViewController.m

```diff
@@ -20,46 +21,58 @@ @interface DealsViewController () <UITableViewDelegate, UITableViewDataSource>
 
 @implementation DealsViewController
 
+-(void)viewDidLoad {
+    [super viewDidLoad];
+    self.tableView.dataSource = self;
+    self.tableView.delegate = self;
+    self.tableView.rowHeight = UITableViewAutomaticDimension;
+    
+}
 - (void)viewWillAppear:(BOOL)animated {
     [super viewWillAppear:YES];
     
-//    [APIManager fetchDealsFromUPCDatabase:self.barcode];
+    [APIManager fetchDealsFromUPCDatabase:self.barcode];
 //    [APIManager fetchDealsFromSearchUPCAPI:self.barcode];
-//    [APIManager fetchDealsFromEbayAPI:self.barcode];
-//    [DatabaseManager fetchItem:self.barcode viewController:self withCompletion:^(NSArray * _Nonnull deals, NSError * _Nonnull error) {
-//        if (deals.count > 0) {
-//            self.deals = (NSMutableArray *) deals;
-////            NSLog(@"%@", self.deals[0]);
-//
-//            AppDeal *deal = self.deals[0];
-//
-//            NSLog(@"%@", deal.price);
-//            [self.tableView reloadData];
-//        }
-//        else {
-//            //alert
-//            NSLog(@"error %@", error.localizedDescription);
-//        }
-//    }];
+//    [APIManager fetchDealsFromEbayAPI:@"53039031"];
+
+    [DatabaseManager fetchItem:self.barcode viewController:self withCompletion:^(NSArray * _Nonnull deals, NSError * _Nonnull error) {
+        if (deals.count > 0) {
+            self.deals = (NSMutableArray *) deals;
+
+            [self.tableView reloadData];
+        }
+        else {
+            //alert
+            NSLog(@"error %@", error.localizedDescription);
+        }
+    }];
 }
 
 - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
     DealCell *cell = [tableView dequeueReusableCellWithIdentifier:@"DealCell"];
     
+    AppDeal *deal = self.deals[indexPath.row];
```

Comment:
As I mentioned early, you need to be careful with arrays. You should always have a check for index validity before you access the value